Dropdown/src/Dropdown.test.js

```diff
@@ -0,0 +1,140 @@
+
+    it("Renders with minimal setup", () => {
+
+        const wrapper = shallow(<Dropdown options={[]} value={null} onSelect={f => f}/>);
+        expect(wrapper).toMatchSnapshot();
+
+    });
+
+    it("Renders labels properly according to props", () => {
+
+        const wrapper = shallow(<Dropdown options={[]} value={null} onSelect={f => f}/>);
+        expect(wrapper).toMatchSnapshot();
+
+        wrapper.setProps({label: "My new label"});
+        const modifiedWrapper = wrapper.update();
+        expect(modifiedWrapper).toMatchSnapshot();
+        let collapsibleLabel = modifiedWrapper.find(".collapsible-label").first();
+
+        expect(collapsibleLabel.text()).toEqual("My new label");
+
+        const secondSelection = "second";
+
+        modifiedWrapper.setProps({
+            options,
+            value: secondSelection,
+            label: "My new label",
+            prependWith: "Prepend text:"
+        });
+        const dropDownWithOptions = modifiedWrapper.update();
+        collapsibleLabel = dropDownWithOptions.find(".collapsible-label").first();
+        expect(dropDownWithOptions).toMatchSnapshot();
+        expect(collapsibleLabel.text()).toEqual(`Prepend text: Second Label`);
+
+    });
+
+    it("Displays a dropdown list onClick", () => {
+
+        const props = {options, value: "second", label: "My new label", prependWith: "Prepend text:"};
+
+        const wrapper = mount(<Dropdown onSelect={f => f} {...props}/>);
+
+        let json = toJson(wrapper);
+        expect(json).toMatchSnapshot();
+
+        expect(wrapper.state('dropDownOpen')).toBe(false);
+
+        let collapsibleLabel = wrapper.find(".collapsible-label").first();
+        collapsibleLabel.simulate("click");
+
+        expect(wrapper.state('dropDownOpen')).toBe(true);
+
+        json = toJson(wrapper.update());
+
+        expect(json).toMatchSnapshot();
+    });
+
+    it("Displays/Hides icon when withIcon is true", () => {
+
+        const props = {options, value: "second", label: "My new label", prependWith: "Prepend text:", withIcon: false};
+        const wrapper = shallow(<Dropdown onSelect={f => f} {...props}/>);
+        expect(wrapper).toMatchSnapshot();
+
+        let icon = wrapper.find(".dropdown-icon");
+        expect(icon.length).toBe(0);
+
+        wrapper.setProps({withIcon: true});
+        wrapper.update();
+        icon = wrapper.find(".dropdown-icon");
+        expect(icon.length).toBe(1);
+        expect(wrapper).toMatchSnapshot();
+
+    });
+
+    it("Renders <Collapse />", () => {
+        const props = {
+            options,
+            value: "second",
+            label: "My new label",
+            prependWith: "Prepend text:",
+            withIcon: false,
+            isDropDownOpen: true
+        };
+        const deepRendering = mount(<Dropdown onSelect={f => f} {...props}/>);
+
+        const collapse = deepRendering.find(Collapse);
+        expect(collapse).toHaveLength(1);
```

Comment:
This is an example of what we discussed previously. A test called `"Renders <Collapse />"` actually ends on this line. Anything that follows deserves separate test case(s). 

Have you discussed with @antoniodgonzalez to use nested `describe` to group similar setup and decompose assertions? In this case I'd probably extract a `beforeEach` and have a test for `Collapse`, a test for the click event and a test for the snapshot.

Reply:
This was a full rendering test made on purpose, since it is a complex components, I wanted to create a snapshot of a real use case. It helped me a lot during by refactoring. 
I agree about splitting but this test case is also helpful since it's very close to a manual test, IMHO.